experimental/Schemes/singular_locus.jl

```diff
@@ -3,28 +3,115 @@ export singular_locus, singular_locus_reduced
 ###       as soon as not only schemes, but also varieties as special 
 ###       schemes have been introduced in OSCAR
-function singular_locus(X::AbsSpec{<:Ring, <:MPAnyQuoRing})
+### TODO: Make singular locus also available for projective schemes and
+###       for covered schemes (the workhorse is already there...).
+ 
+@doc Markdown.doc"""
+    singular_locus(X::AbsSpec{<:Field, <:MPolyAnyRing}) -> AbsSpec
+
+Return the singular locus of `X`.
+
+For computing the singular locus of the reduced scheme induced by `X`, please use [`singular_locus_reduced`](@ref).
+
+Currently this command is available for affine schemes and space germs.
+TODO: Covered schemes, projective schemes
+
+Over non-perfect fields, this command returns the non-smooth locus and `X`
+may still be regular at some points of the returned subscheme.
+
+# Examples
+``` jldoctest
+julia> R, (x,y,z) = QQ["x", "y", "z"]
+(Multivariate Polynomial Ring in x, y, z over Rational Field, fmpq_mpoly[x, y, z])
+
+julia> I = ideal(R, [x^2 - y^2 + z^2])
+ideal(x^2 - y^2 + z^2)
+
+julia> A3 = Spec(R)
+Spec of Multivariate Polynomial Ring in x, y, z over Rational Field
+
+julia> X = Spec(R,I)
+Spec of Quotient of Multivariate Polynomial Ring in x, y, z over Rational Field by ideal(x^2 - y^2 + z^2)
+
+julia> singular_locus(A3)
+Spec of Quotient of Multivariate Polynomial Ring in x, y, z over Rational Field by ideal(1)
+
+julia> singular_locus(X)
+Spec of Quotient of Multivariate Polynomial Ring in x, y, z over Rational Field by ideal(x^2 - y^2 + z^2, 2*x, -2*y, 2*z)
+
+julia> U = MPolyComplementOfKPointIdeal(R,[0,0,0])
+complement of maximal ideal corresponding to point with coordinates fmpq[0, 0, 0]
+
+julia> Y = Spec(R,I,U)
+Spec of Localization of Quotient of Multivariate Polynomial Ring in x, y, z over Rational Field by ideal(x^2 - y^2 + z^2) at the multiplicative set complement of maximal ideal corresponding to point with coordinates fmpq[0, 0, 0]
+
+julia> singular_locus(Y)
+Spec of Localization of Quotient of Multivariate Polynomial Ring in x, y, z over Rational Field by ideal(x^2 - y^2 + z^2, x^2 - y^2 + z^2, 2*x, -2*y, 2*z, x^2 - y^2 + z^2, x^2 - y^2 + z^2) at the multiplicative set complement of maximal ideal corresponding to point with coordinates fmpq[0, 0, 0]
+
+```
+"""
+function singular_locus(X::AbsSpec{<:Field, <:MPAnyQuoRing})
   comp = _singular_locus_with_decomposition(X,false)
   if length(comp) == 0 
     return subscheme(X, ideal(OO(X),one(OO(X))))
   end
   R = base_ring(OO(X))
   I= prod([modulus(quotient_ring(OO(Y))) for Y in comp])
-#  return Spec(R, I, inverted_set(OO(X)))
   return subscheme(X,I)
 end
 
+# make singular_locus agnostic to quotient
+function singular_locus(X::AbsSpec{<:Field, <:MPAnyNonQuoRing})
+  return subscheme(X,ideal(OO(X),[one(OO(X))]))
+end
+
+@doc Markdown.doc"""
+    singular_locus_reduced(X::AbsSpec{<:Field, <:MPolyAnyRing}) -> AbsSpec
+
+Return the singular locus of the reduced scheme 'X_{red}' induced by `X`.
```

Comment:
If you want to render this a latex the syntax is:
```suggestion
Return the singular locus of the reduced scheme ``X_{red}`` induced by `X`.
```

Reply:
Thank you. Done.